File: Problems/No.1753_Maximum_Score_From_Removing_Stones/maximum_score_from_removing_stones.py
class Solution(object):
    def maximumScore(self, a, b, c):
        """
        :type a: int
        :type b: int
        :type c: int
        :rtype: int
        """

        # parameters
        StoneAmounts = []
        moves = 0

        # start computing
        StoneAmounts = [-a, -b, -c]
        heapq.heapify(StoneAmounts)
        while StoneAmounts:
            stone1 = heapq.heappop(StoneAmounts)
            stone2 = heapq.heappop(StoneAmounts)
            if ((stone1 == 0) or (stone2 == 0)):
                break
            else:
                moves += 1
                heapq.heappush(StoneAmounts, stone1+1)
                heapq.heappush(StoneAmounts, stone2+1)

        return moves

File: Problems/No.1753_Maximum_Score_From_Removing_Stones/maximum_score_from_removing_stones.cpp
class Solution {
public:
    int maximumScore(int a, int b, int c) {
        
        // parameters
        std::priority_queue<int> StoneAmounts;
        int moves = 0;

        // start computing
        StoneAmounts.push(a);
        StoneAmounts.push(b);
        StoneAmounts.push(c);
        while (!StoneAmounts.empty()) {

            auto stone1 = StoneAmounts.top();
            StoneAmounts.pop();
            auto stone2 = StoneAmounts.top();
            StoneAmounts.pop();

            if ((stone1 == 0) || (stone2 == 0)) break;
            else {

                moves++;
                StoneAmounts.push(stone1-1);
                StoneAmounts.push(stone2-1);

            }

        }

        return moves;

    }
};
